runtime/src/main/java/org/corfudb/runtime/LogReplicationUtils.java

```diff
@@ -1,14 +1,269 @@
+                CorfuStoreEntry<ReplicationStatusKey, ReplicationStatusVal, Message> entry = entries.get(0);
+                txn.commit();
+                return entry.getPayload().getDataConsistent() ? false : true;
+            } catch (TransactionAbortedException e) {
+                // Since this is a read-only transaction, the abort can only be if this version of the table is not
+                // available in the JVM's cache.  The underlying cause for this error is TrimmedException
+                Preconditions.checkState(e.getCause() instanceof TrimmedException);
+
+                // This means that there have been updates to the Replication Status table after the above timestamp,
+                // and this snapshot has been evicted from the cache of the JVM.  We cannot tell the status of snapshot
+                // sync at this timestamp.  Return true so that caller retries the check with a later timestamp.
+                return true;
+            }
+        }
+
+        /**
+         * Wait for the snapshot sync to complete.  The completion is signalled when SnapshotSyncCompletionListener
+         * invokes notifyAll().
+         * @param timestamp
+         */
+        private void waitSnapshotSyncCompletion(CorfuStoreMetadata.Timestamp timestamp) {
+            SnapshotSyncCompletionListener snapshotSyncCompletionListener = new SnapshotSyncCompletionListener(timestamp);
+            corfuStore.subscribeListener(snapshotSyncCompletionListener, CORFU_SYSTEM_NAMESPACE, LR_STATUS_STREAM_TAG,
+                    Arrays.asList(REPLICATION_STATUS_TABLE), timestamp);
+
+            synchronized (timestamp) {
+                while (!snapshotSyncCompletionListener.isSnapshotSyncComplete()) {
+                    try {
+                        timestamp.wait();
+                    } catch (Exception e) {
+                        log.error("Exception while waiting for snapshot sync to complete", e);
+                        corfuStore.unsubscribeListener(snapshotSyncCompletionListener);
+                        throw new RuntimeException(e);
+                    }
+                }
+                corfuStore.unsubscribeListener(snapshotSyncCompletionListener);
+            }
+        }
+    }
+
+    private class SnapshotSyncCompletionListener implements StreamListener {
+
+        private CorfuStoreMetadata.Timestamp subscriptionTimestamp;
+
+        @Getter
+        private boolean snapshotSyncComplete;
+
+        SnapshotSyncCompletionListener(CorfuStoreMetadata.Timestamp subscriptionTimestamp) {
+            this.subscriptionTimestamp = subscriptionTimestamp;
+            this.snapshotSyncComplete = false;
+        }
+
+        @Override
+        public void onNext(CorfuStreamEntries results) {
+            Map<TableSchema, List<CorfuStreamEntry>> entries = results.getEntries();
+            for (TableSchema tableSchema : entries.keySet()) {
+                if (tableSchema.getTableName().equals(REPLICATION_STATUS_TABLE)) {
```

Comment:
Do we need this 'if' ? I think updates to replication_status happens in a separate txn?

Reply:
the 'if' is not to filter out data writes.  If we add other LR tables in future with the same stream tag as that used in the Status table, we can ignore those updates.

Reply:
removed the if check as discussed offline